### PR TITLE
Add SA and RoleBinding for ASN ETL pipeline

### PR DESCRIPTION
**Pull request by BobyMCbobs**
Ensures that the asn-etl ServiceAccount is managed using k8s infra Terraform.
Adds a RoleBinding to ensure that the asn-etl ServiceAccount has access to the pii bucket for logs to consume.

### CODE REVIEW

File: infra/gcp/terraform/k8s-infra-ii-sandbox/main.tf [hunk 1]
```diff
@@ -69,3 +69,9 @@ resource "google_bigquery_dataset_iam_member" "bq_data_transfer_binding" {
   role       = "roles/bigquery.admin"
   member     = "serviceAccount:${data.google_service_account.bq_data_transfer.email}"
 }
+
+resource "google_service_account" "asn_etl" {
```
Comment (ameukam) [inline infra/gcp/terraform/k8s-infra-ii-sandbox/main.tf:73]: @BobyMCbobs early review; Can we have a similar service account on https://github.com/kubernetes/k8s.io/tree/main/infra/gcp/terraform/k8s-infra-public-pii ?

Comment (BobyMCbobs) [reply infra/gcp/terraform/k8s-infra-ii-sandbox/main.tf:73]: Thank you @ameukam for taking a look at this PR.

What would the ServiceAccount be called and used for?

Comment (ameukam) [reply infra/gcp/terraform/k8s-infra-ii-sandbox/main.tf:73]: You can keep the same name. I would to see any new change done in this project mirrored in `k8s-infra-public-pii`.

Comment (BobyMCbobs) [reply infra/gcp/terraform/k8s-infra-ii-sandbox/main.tf:73]: Right! That will make it easy for having it moved over to production when fully ready.

Comment (BobyMCbobs) [reply infra/gcp/terraform/k8s-infra-ii-sandbox/main.tf:73]: OK, made the change in a556ada1

File: infra/gcp/terraform/k8s-infra-public-pii/main.tf [hunk 2]
```diff
@@ -85,6 +85,25 @@ resource "google_storage_bucket" "audit-logs-gcs" {
...
+  project      = local.project_id
+  account_id   = "asn-etl"
+  display_name = "asn-etl"
+}
+
+// service account for running ASN etl pipeline job
+data "google_service_account" "public_pii_asn_etl" {
+  account_id = "asn-etl"
+  project    = local.project_id
+}
```
Comment (ameukam) [inline infra/gcp/terraform/k8s-infra-public-pii/main.tf:99]: Not needed. The resource `google_service_account.asn_etl` create the service account.

File: infra/gcp/terraform/k8s-infra-public-pii/main.tf [hunk 3]
```diff
@@ -85,6 +85,25 @@ resource "google_storage_bucket" "audit-logs-gcs" {
...
+
+// service account for running ASN etl pipeline job
+data "google_service_account" "ii_sandbox_asn_etl" {
+  account_id = "asn-etl"
+  project    = "k8s-infra-ii-sandbox"
+}
```
Comment (ameukam) [inline infra/gcp/terraform/k8s-infra-public-pii/main.tf:105]: Not needed. Only workloads or applications running inside this project are allowed to interact with BigQuery.

File: infra/gcp/terraform/k8s-infra-public-pii/main.tf [hunk 4]
```diff
@@ -99,6 +118,18 @@ data "google_iam_policy" "audit_logs_gcs_bindings" {
       "group:cloud-storage-analytics@google.com",
     ]
   }
+  binding {
+    role = "roles/storage.legacyBucketReader"
+    members = [
+      "serviceAccount:${data.google_service_account.public_pii_asn_etl}",
```
Comment (ameukam) [inline infra/gcp/terraform/k8s-infra-public-pii/main.tf:124]:
```suggestion
      "serviceAccount:${google_service_account.asn_etl.email}",
```

File: infra/gcp/terraform/k8s-infra-public-pii/main.tf (hunk 2 above)
Comment (BobyMCbobs) [reply infra/gcp/terraform/k8s-infra-public-pii/main.tf:99]: This is used for accessing referring to the ServiceAccount in the RoleBindings below.

File: infra/gcp/terraform/k8s-infra-public-pii/main.tf (hunk 3 above)
Comment (BobyMCbobs) [reply infra/gcp/terraform/k8s-infra-public-pii/main.tf:105]: Using this to refer to the account in the RoleBinding below

File: infra/gcp/terraform/k8s-infra-public-pii/main.tf (hunk 4 above)
Comment (BobyMCbobs) [reply infra/gcp/terraform/k8s-infra-public-pii/main.tf:124]: Ah, this is better, I'll update it

File: infra/gcp/terraform/k8s-infra-public-pii/main.tf [hunk 5]
```diff
@@ -99,6 +112,18 @@ data "google_iam_policy" "audit_logs_gcs_bindings" {
...
+    members = [
+      "serviceAccount:${google_service_account.public_pii_asn_etl.email}",
+    ]
+  }
+  binding {
+    role = "roles/storage.legacyBucketReader"
+    members = [
+      "serviceAccount:${data.google_service_account.ii_sandbox_asn_etl.email}",
+    ]
+  }
```
Comment (ameukam) [inline infra/gcp/terraform/k8s-infra-public-pii/main.tf:126]: I'm not sure to understand by why do we need to do this binding ? Is it possible to have something that explain the different interactions between BigQuery and Google Cloud Storage ?

Comment (BobyMCbobs) [reply infra/gcp/terraform/k8s-infra-public-pii/main.tf:126]: The Pii data is in a bucket named _k8s-infra-artifacts-gcslogs_ in the _public_pii_ project.
In order to run the job to generate up to date data to display for the image/artifact spending DataStudio report, the logs inside of the bucket are needed.
The ServiceAccount that the job runs as needs to have access to the bucket mentioned.
After discussing with @spiffxp and @hh, I believe these are the right lines to add.

Comment (ameukam) [reply infra/gcp/terraform/k8s-infra-public-pii/main.tf:126]: SGTM

**Comment by BobyMCbobs** [comment]
I think I'm ready for a merge
/assign @ameukam

**Comment by ameukam** [comment]
I'll defer to @spiffxp for approvaL.

**Comment by BobyMCbobs** [comment]
> /lgtm
> /approve

Woohoo! Thank you @ameukam!

**Comment by ameukam** [comment]
@BobyMCbobs I would to see the prowjob that use this service account in https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/wg-k8s-infra so we don't have to transfer from one project to another.

**Comment by ameukam** [comment]
The serviceaccount already exists. 

```
google_service_account.asn_etl: Creating...
╷
│ Error: Error creating service account: googleapi: Error 409: Service account asn-etl already exists within project projects/k8s-infra-ii-sandbox., alreadyExists
│ 
│   with google_service_account.asn_etl,
│   on main.tf line 73, in resource "google_service_account" "asn_etl":
│   73: resource "google_service_account" "asn_etl" {
```

**Comment by BobyMCbobs** [comment]
> @BobyMCbobs I would to see the prowjob that use this service account in https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/wg-k8s-infra so we don't have to transfer from one project to another.

Sounds good to me!

**Comment by BobyMCbobs** [comment]
> The serviceaccount already exists.
> 
> ```
> google_service_account.asn_etl: Creating...
> ╷
> │ Error: Error creating service account: googleapi: Error 409: Service account asn-etl already exists within project projects/k8s-infra-ii-sandbox., alreadyExists
> │ 
> │   with google_service_account.asn_etl,
> │   on main.tf line 73, in resource "google_service_account" "asn_etl":
> │   73: resource "google_service_account" "asn_etl" {
> ```

It is important that the ServiceAccount to be moved into being managed as it was created manually originally. Do you have any thoughts?

**Comment by ameukam** [comment]
> > The serviceaccount already exists.
> > ```
> > google_service_account.asn_etl: Creating...
> > ╷
> > │ Error: Error creating service account: googleapi: Error 409: Service account asn-etl already exists within project projects/k8s-infra-ii-sandbox., alreadyExists
> > │ 
> > │   with google_service_account.asn_etl,
> > │   on main.tf line 73, in resource "google_service_account" "asn_etl":
> > │   73: resource "google_service_account" "asn_etl" {
> > ```
> 
> It is important that the ServiceAccount to be moved into being managed as it was created manually originally. Do you have any thoughts?

You can't create a Service Account already existing with terraform. I imported the service Account in the terraform state with `terraform import`. A `terraform plan -refresh=true` should be good now.

**Comment by ameukam** [comment]
Typo in `k8s-infra-public-pii` : 

```shell
❯ tf plan -refresh=true
╷
│ Error: Reference to undeclared resource
│ 
│   on main.tf line 118, in data "google_iam_policy" "audit_logs_gcs_bindings":
│  118:       "serviceAccount:${google_service_account.public_pii_asn_etl.email}",
│ 
│ A managed resource "google_service_account" "public_pii_asn_etl" has not been declared in the root module.
╵
```

Fixed in https://github.com/kubernetes/k8s.io/pull/2605

**Comment by BobyMCbobs** [comment]
> > > The serviceaccount already exists.
> > > ```
> > > google_service_account.asn_etl: Creating...
> > > ╷
> > > │ Error: Error creating service account: googleapi: Error 409: Service account asn-etl already exists within project projects/k8s-infra-ii-sandbox., alreadyExists
> > > │ 
> > > │   with google_service_account.asn_etl,
> > > │   on main.tf line 73, in resource "google_service_account" "asn_etl":
> > > │   73: resource "google_service_account" "asn_etl" {
> > > ```
> > 
> > 
> > It is important that the ServiceAccount to be moved into being managed as it was created manually originally. Do you have any thoughts?
> 
> You can't create a Service Account already existing with terraform. I imported the service Account in the terraform state with `terraform import`. A `terraform plan -refresh=true` should be good now.

Good stuff, appreciate it!

**Pull request merged**